### ISSUE TITLE
Radius Server

### ISSUE THREAD

**Issue by vamadir** [clowwindy/ShadowVPN#159]
Did you have solution to use ShadowVPN with Radius Server like pptpd or l2tpd??

**Comment by Liashui** [comment]
@vamadir   Did you find solution to auth users with radius server,I suffer a lot on this issue

**Comment by vamadir** [comment]
@Liashui No, i didn`t find any solution with radius server. But ss-server can work with ss-manager, what you can use for multiuser connection. Or write some software for database connection. 
I am not a programmer, so i just waiting for some realization.
Now my server working ss-server <-> ss-manager <-> multiuser`s

**Comment by Liashui** [comment]
i see this in server.conf

# Users allowed. Each must be HEX of 8 bytes. You can generate on by running:
#     xxd -l 8 -p /dev/random
# See `net` for more information.
# user_token=7e335d67f1dc2c01,ff593b9e6abeb2a5,e3c7b8db40a96105

it seems support muti users,but the file only read once when vpn(server mode) start,if you use it commercial,u will probably register clients

i am trying to replace usertoken in server.conf with usertoken in mysql,but i am not sure is this way popular

**Comment by Liashui** [comment]
@vamadir

**Comment by vamadir** [comment]
@Liashui i didnt know about shadowsocks-manager project correct. But when i try it its be not stable. Maybe now something change.

For multiusers you can do like this:
make server but with ss-manager (of shadowsocks-libev) at startup config(ss-manager) you writing --executable _ss-server_file_ 

```
[Service]
Type=simple
User=root
Restart=on-abort
ExecStart=/usr/bin/ss-manager --manager-address /tmp/shadowsocks-manager.sock --executable /usr/bin/ss-server -a $USER -c /etc/shadowsocks-libev/config-multiuser.json

[Install]
WantedBy=multi-user.target
```
Now you can connect to /tmp/shadowsocks-manager.sock from local
or you can change /tmp/shadowsocks-manager.sock to _YOUR_IP_:_YOUR_PORT_
Then you can connect from outside. 
And for add or change users you no need config file. Its all making on fly. when you connect to ss-manager. https://github.com/shadowsocks/shadowsocks/wiki/Manage-Multiple-Users
also you can read man ss-manager (on linux)

P.S.
ss-manager create hided configs at home directory. Home directory is user of ss-manager (root default)

**Comment by Liashui** [comment]
@vamadir  shadowsock server support what you mention abve on some particular version, shdowvpn donot.

**Comment by vamadir** [comment]
@sorry, i forgot, this is shadowvpn topic not shadowssocks-libev

Shadowvpn, i didnt find any multuuser solution
